sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs

```diff
@@ -130,17 +130,11 @@ public override int PrefetchCount
         /// </summary>
         internal readonly ConcurrentExpiringSet<Guid> RequestResponseLockedMessages;
 
-        private static IReadOnlyList<ServiceBusReceivedMessage> s_backingEmptyList;
         private readonly bool _isProcessor;
 
-        private static IReadOnlyList<ServiceBusReceivedMessage> EmptyList
```

Comment:
@danielmarbach is this change as performant as what we had previously? I'm assuming it is as we are still not doing allocations if the empty list is not needed. But it seems this should be a slight improvement in the event that the empty list _is_ needed. Though I acknowledge I may be overlooking something.

Reply:
It should be.  

The main difference is that we're paying an upfront cost to eagerly allocate an empty array (once per process, as instances are shared by `Array`) rather than deferred double allocation (list/read-only wrapper) locally.

Reply:
Good point, we are doing the allocation once upfront now, but it will still be cheaper than the alternative if we end up needing the empty list.

Reply:
Agreed. Had a moment to see why you need the backing field instead of directly using Array.Empty.

Reply:
@jsquire @JoshLove-msft https://github.com/Azure/azure-amqp/pull/255 has been merged to the hotfix branch